### ISSUE TITLE
Rooms: [Windows] Instances copied using room_assign() do not perform their create event in the target room

### ISSUE THREAD

**Issue by iampremo** [YoYoGames/GameMaker-Bugs#306]
GM-1210

When a room's content is copied into another room using room_assign(index, source) and the source room contains an object with a create event, the object's create event does not play in the target room. This can cause issues when the create event contains variable definitions that are being used in other events.

Expected results: 

The create event of objects copied from room to room using room_assign() runs correctly in the target room.

 

Steps:
 # Run IDE and import project attached
 # Run project
 # Press Space
 # Observe issue.

**Comment by TheouAegis** [comment]
tldr; the bug still exists with room_assign()

This was a known issue for 5 years now (Reddit post, allegedly was reported to YYG back then). When going to a room modified by room_assign(), all the instances in the new room are duplicated with identical ids. That is, suppose a room has instances 100012 and 100013; if room_assign() was called, the game will have two 100012 and two 100013 in the room. 

Now let's say 100013 runs first. Typically, the _real_ instance will get to run first, from what I've noticed. Then the fake instance will run its code. However, since the fake instance does not exist, any user-created variables (built-in variables seem to work) will cause the game to crash when read. 

Additionally, if there is only one instance in the original room, the instance's variables can be read _until_ the after the End Step. By the Draw event, the custom variables are all unreadable. 

As of GMS2024, it seems the struct created when defining variables is only applied to the real instance. The cloned instances do not have access to that struct, which is why built-in variables work (or maybe they don't, but they don't cause errors either). 

My hunch is the code for copying instances from one room to the next used a permanent variable, rather than a temporary variable somewhere.
